patches/tModLoader/Terraria/UI/AchievementAdvisor.cs.patch

```diff
@@ -0,0 +1,34 @@
+--- src/TerrariaNetCore/Terraria/UI/AchievementAdvisor.cs
++++ src/tModLoader/Terraria/UI/AchievementAdvisor.cs
+@@ -1,9 +_,11 @@
++using System;
+ using System.Collections.Generic;
+ using System.Linq;
+ using Microsoft.Xna.Framework;
+ using Microsoft.Xna.Framework.Graphics;
+ using ReLogic.Content;
+ using Terraria.GameInput;
++using Terraria.ModLoader;
+ 
+ namespace Terraria.UI;
+ 
+@@ -146,7 +_,18 @@
+ 
+ 		Rectangle frame = card.frame;
+ 		frame.X += 528;
++		if (card.achievement.Modded) {
++			if (card.achievement.ModAchievement.GetTexture() == null) {
++				throw new Exception($"{card.achievement.Name}.png was not found, add it in the same directory as your source file.");
++			}
++		}
```

Comment:
This should be checked during `SetupContent` after which we can assume that `Texture != null`